controls/ubuntu-18.04-server-cis-4.1.10.rb

```diff
@@ -130,14 +130,39 @@
   tag cis_cdc_version: "7"
   tag cis_rid: "4.1.10"
 
-  describe auditd do
-    its('lines') { should include "-a always,exit -F arch=b32 -S creat -S open -S openat -S truncate -S ftruncate -F exit=-EACCES -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k access" }
-    its('lines') { should include "-a always,exit -F arch=b32 -S creat -S open -S openat -S truncate -S ftruncate -F exit=-EPERM -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k access" }
-  end
+  arches = ['b32']
   if os.arch.match?(/64/)
-    describe auditd do
-      its('lines') { should include "-a always,exit -F arch=b64 -S creat -S open -S openat -S truncate -S ftruncate -F exit=-EACCES -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k access" }
-      its('lines') { should include "-a always,exit -F arch=b64 -S creat -S open -S openat -S truncate -S ftruncate -F exit=-EPERM -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k access" }
+    arches.push('b64')
+  end
+
+  syscalls = [
+    'creat',
+    'open',
+    'openat',
+    'truncate',
+    'ftruncate'
+  ]
+
+  syscalls.each do |syscall|
+    arches.each do |a|
+      describe auditd.syscall(syscall).where { arch == a && key == 'access'  } do
+        its('action.uniq') { should eq ['always'] }
+        its('list.uniq') { should eq ['exit'] }
+        its('fields.flatten.uniq') {  should include "auid>=#{login_defs.UID_MIN}" }
+        its('exit') { should include '-EACCES' }
+        its('exit') { should include '-EPERM' }
+      end
+
+      # the rule states that `auid!=4294967295` but this comes back as `auid!=-1` when queries via `auditctl -l`
```

Comment:
this little bit threw me for a bit of a loop. It might be safe to just compare against `auid=-1` but I fear that doing that might be brittle, because I'm not sure I understand why it works this way, or if it can be relied upon to work like that.

From my understanding, the audit system treats `auid` as an unsigned integer, so `-1` and `4294967295` are two interpretations of the same representation: `0xFFFFFFFF` (which is the case when a daemon makes a syscall, so there is no `auid`). But, at the point we are dealing with this value here, as a 'field', it is a string value.